webviz_subsurface/_utils/vector_calculator.py

```diff
@@ -0,0 +1,233 @@
+
+    return True, ""
+
+
+def variable_vector_map_from_dict(
+    variable_vector_dict: Dict[str, str]
+) -> List[VariableVectorMapInfo]:
+    variable_vector_map: List[VariableVectorMapInfo] = []
+    for variable in variable_vector_dict:
+        variable_vector_map.append(
+            {
+                "variableName": variable,
+                "vectorName": [variable_vector_dict[variable]],
+            }
+        )
+    return variable_vector_map
+
+
+def expressions_from_config(
+    expressions: Dict[str, ConfigExpressionData],
+) -> List[ExpressionInfo]:
+    output: List[ExpressionInfo] = []
+    for expression in expressions:
+        output.append(
+            {
+                "name": expression,
+                "expression": expressions[expression]["expression"],
+                "id": f"{uuid4()}",
+                "variableVectorMap": variable_vector_map_from_dict(
+                    expressions[expression]["variableVectorMap"]
+                ),
+                "isValid": False,  # Set default false? And expect seperate validation?
+                "isDeletable": False,
+            }
+        )
+    return output
+
+
+def is_vector_name_existing(name: str, vector_data: list) -> bool:
+    nodes = name.split(":")
+    current_child_list = vector_data
+    for node in nodes:
+        found = False
+        for child in current_child_list:
+            if child["name"] == node:
+                children = child["children"]
+                current_child_list = children if children is not None else []
+                found = True
+                break
+        if not found:
+            return False
+    return found
+
+
+@CACHE.memoize(timeout=CACHE.TIMEOUT)
+def get_selected_expressions(
+    expressions: List[ExpressionInfo], selected_names: List[str]
+) -> List[ExpressionInfo]:
+    selected: List[ExpressionInfo] = []
+    for name in selected_names:
+        selected_expression: ExpressionInfo = next(
+            (elm for elm in expressions if elm["name"] == name), None
+        )
+        if selected_expression is not None and selected_expression not in selected:
+            selected.append(selected_expression)
+    return selected
+
+
+@CACHE.memoize(timeout=CACHE.TIMEOUT)
+def get_calculated_vectors(
+    expressions: List[ExpressionInfo],
+    smry: pd.DataFrame,
+) -> pd.DataFrame:
+    calculated_vectors: pd.DataFrame = pd.DataFrame()
+    for elm in expressions:
+        name: str = elm["name"]
+        expr: str = elm["expression"]
+        var_vec_dict: Dict[str, str] = VectorCalculator.variable_vector_dict(
+            elm["variableVectorMap"]
+        )
+
+        values: Dict[str, pd.Series] = {}
+        for var in var_vec_dict:
+            values[var] = smry[var_vec_dict[var]]
+
+        evaluated_expr = VectorCalculator.evaluate_expression(expr, values)
+        if evaluated_expr is not None:
+            calculated_vectors[name] = evaluated_expr
+    return calculated_vectors
```

Comment:
From video discussion:
```suggestion
def get_calculated_vectors(
    expressions: List[ExpressionInfo],
    smry: pd.DataFrame,
) -> pd.DataFrame:
    calculated_vectors: pd.DataFrame = pd.DataFrame()
    for elm in expressions:
        name: str = elm["name"]
        expr: str = elm["expression"]
        var_vec_dict: Dict[str, str] = VectorCalculator.variable_vector_dict(
            elm["variableVectorMap"]
        )

        values: Dict[str, np.ndarray] = {}
        for var in var_vec_dict:
            values[var] = smry[var_vec_dict[var]].values

        evaluated_expr = VectorCalculator.evaluate_expression(expr, values)
        if evaluated_expr is not None:
            calculated_vectors[name] = evaluated_expr
    return calculated_vectors
```